plugins/modules/database.py

```diff
@@ -13,74 +12,83 @@
 version_added: "1.0.0"
 author: "Omer Ratsaby <omer.ratsaby@ravendb.net> (@thegoldenplatypus)"
 
 extends_documentation_fragment:
-- ravendb.ravendb.ravendb
+  - ravendb.ravendb.ravendb
 
 options:
-    replication_factor:
-        description:
-            - Number of server nodes to replicate the database to.
-            - Must be a positive integer.
-            - Only used when creating a database.
-        required: false
-        default: 1
-        type: int
-    state:
-        description:
-            - Desired state of the database.
-            - If C(present), the database will be created if it does not exist.
-            - If C(absent), the database will be deleted if it exists.
-        required: false
-        type: str
-        choices:
-          - present
-          - absent
-        default: present
-    encrypted:
-        description:
-        - Create the database as encrypted.
-        - When C(true), the module ensures a secret key is assigned (generated or read from file) and distributed to all cluster nodes before creation.
-        - Requires C(certificate_path) to access admin endpoints.
-        required: false
-        default: false
-        type: bool
-    encryption_key:
-        description:
-        - Path to a file that contains the raw encryption key (plain text).
-        - Mutually exclusive with C(generate_encryption_key).
-        - Used only when C(encrypted=true).
-        required: false
-        type: str
-    generate_encryption_key:
-        description:
-        - If C(true), asks the server to generate a new encryption key via the admin API.
-        - Mutually exclusive with C(encryption_key).
-        - Used only when C(encrypted=true).
-        required: false
-        default: false
-        type: bool
-    encryption_key_output_path:
-        description:
-        - When C(generate_encryption_key=true), write the generated key to this local file with safe permissions (0600 umask).
-        - Ignored if C(generate_encryption_key=false).
-        required: false
-        type: str
-    database_settings:
-        description:
-          - Dictionary of database-level settings to apply.
-          - Values are normalized to strings and compared against current customized settings.
-          - If differences exist, the module updates settings and toggles the database state to reload them safely.
-        required: false
-        type: dict
-        default: {}
+  replication_factor:
+    description:
+      - Number of server nodes to replicate the database to.
+      - Must be a positive integer.
+      - Only used when creating a database.
+    required: false
+    default: 1
+    type: int
+  topology_members:
+    description:
+      - Optional list of cluster node tags to host this database (fixed placement).
+      - When provided, its length must equal C(replication_factor).
+    required: false
+    type: list
+    elements: str
+    default: []
```

Comment:
again, we need to validate if at least one of the original nodes is present in the resulting set OR wait for replication before deletion from the original nodes